tests/test_modules/test_contact.py

```diff
@@ -0,0 +1,274 @@
+        contact = load_contact
+        contact.using(legal_contact, 3, 4.0, "5", k=1, l_var="2", j=3.0)
+
+        assert contact._contact_cls == legal_contact
+        assert contact._args == (3, 4.0, "5")
+        assert contact._kwargs == {"k": 1, "l_var": "2", "j": 3.0}
+
+    def test_id(self, load_contact):
+        contact = load_contact
+        # This is purely for coverage purposes, no actual test
+        # since its a simple return
+        assert contact.id() == (15, 23)
+
+    def test_call_without_setting_contact_throws_runtime_error(self, load_contact):
+        contact = load_contact
+
+        with pytest.raises(RuntimeError) as excinfo:
+            contact()
+        assert "No contacts provided" in str(excinfo.value)
+
+    def test_call_improper_args_throws(self, load_contact):
+        # Example of bad initiailization function
+        # This needs at least four args which the user might
+        # forget to pass later on
+        def mock_init(self, *args, **kwargs):
+            self.nu = args[3]  # Need at least four args
+            self.k = kwargs.get("k")
+
+        # in place class
+        MockContact = type(
+            "MockContact", (self.NoContact, object), {"__init__": mock_init}
+        )
+
+        # The user thinks 4.0 goes to nu, but we don't accept it because of error in
+        # construction og a Contact class
+        contact = load_contact
+        contact.using(MockContact, 4.0, k=1, l_var="2", j=3.0)
+
+        # Actual test is here, this should not throw
+        with pytest.raises(TypeError) as excinfo:
+            _ = contact()
+        assert "Unable to construct" in str(excinfo.value)
+
+
+class TestContactMixin:
+    from elastica.modules import BaseSystemCollection
+
+    class SystemCollectionWithContactMixedin(BaseSystemCollection, Contact):
+        pass
+
+    from elastica.rod import RodBase
+    from elastica.rigidbody import RigidBodyBase
+    from elastica.surface import SurfaceBase
+
+    class MockRod(RodBase):
+        def __init__(self, *args, **kwargs):
+            self.n_elems = 3  # arbitrary number
+
+        # Contacts assume that this promise is met
+        def __len__(self):
+            return 2  # a random number
+
+    class MockRigidBody(RigidBodyBase):
+        def __init__(self, *args, **kwargs):
+            self.n_elems = 1
+
+        # Contacts assume that this promise is met
+        def __len__(self):
+            return 2  # a random number
+
+    class MockSurface(SurfaceBase):
+        def __init__(self, *args, **kwargs):
+            self.n_facets = 1
+
+        # Contacts assume that this promise is met
+        def __len__(self):
+            return 2  # a random number
+
+    @pytest.fixture(scope="function", params=[2, 10])
+    def load_system_with_contacts(self, request):
+        n_sys = request.param
+        sys_coll_with_contacts = self.SystemCollectionWithContactMixedin()
```

Comment:
```suggestion
        sys_coll_with_contacts = self.SystemCollectionWithContactMixin()
```
?

Reply:
Should I change it in test_connections.py too? it is like that there.

Reply:
I think mixin is the correct usage. Please correct in both.